samples/memory/agentcore_memory_store_long_term_search.ipynb

```diff
@@ -0,0 +1,437 @@
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import LangGraph and LangChain components\n",
+    "from langchain.chat_models import init_chat_model\n",
+    "from langgraph.prebuilt import create_react_agent\n",
+    "from langchain_core.messages import HumanMessage, AIMessage\n",
+    "from langchain_core.runnables import RunnableConfig\n",
+    "from langgraph.store.base import BaseStore\n",
+    "import uuid\n",
+    "import logging\n",
+    "logging.getLogger().setLevel(logging.DEBUG)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import the AgentCoreMemoryStore that we will use as a store\n",
+    "from langgraph_checkpoint_aws import (\n",
+    "    AgentCoreMemoryStore\n",
+    ")\n",
+    "\n",
+    "# For this example, we will just use an InMemorySaver to save context.\n",
+    "# In production, we highly recommend the AgentCoreMemorySaver as a checkpointer\n",
+    "# which works seamlessly alongside the memory store\n",
+    "from langgraph.checkpoint.memory import InMemorySaver"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## AgentCore Memory Configuration\n",
+    "- `REGION` corresponds to the AWS region that your resources are present in, these are passed to the `AgentCoreMemorySaver`.\n",
+    "- `MEMORY_ID` corresponds to your top level AgentCore Memory resource. Within this resource we will store checkpoints for multiple actors and sessions\n",
+    "- `MODEL_ID` this is the bedrock model that will power our LangGraph agent through Bedrock Converse.\n",
+    "\n",
+    "We will use the `MEMORY_ID` and any additional boto3 client keyword args (in our case, `REGION`) to instantiate our checkpointer."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "REGION = \"us-west-2\"\n",
+    "MEMORY_ID = \"YOUR_MEMORY_ID\"\n",
+    "MODEL_ID = \"us.anthropic.claude-3-7-sonnet-20250219-v1:0\"\n",
+    "\n",
+    "# Initialize the store to enable long term memory saving and retrieval\n",
+    "store = AgentCoreMemoryStore(memory_id=MEMORY_ID, region_name=REGION)\n",
+    "\n",
+    "# Initialize Bedrock LLM\n",
+    "llm = init_chat_model(MODEL_ID, model_provider=\"bedrock_converse\", region_name=REGION)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Storing the messages that we want for long term memory extraction\n",
+    "\n",
+    "In Bedrock AgentCore, each conversational message that is added for an `actor_id` and `session_id` is processed in the service according to the strategies specified for that memory resource. In our case, since we defined two strategies, user preference and semantic extraction, both user preferences and facts from previous conversations will be stored as long term memories in the service that can be retrieved by our retrieval tool above.\n",
```

Comment:
```suggestion
    "In Bedrock AgentCore, each conversational message that is added for an `actor_id` and `session_id` is processed in the service according to the strategies specified for that memory resource. In our case, since we defined two strategies, user preference and semantic extraction, both user preferences and facts from previous conversations will be stored as long term memories in the service that can be retrieved by the application.\n",
```